holidays/countries/brunei.py

```diff
@@ -156,61 +192,75 @@ def _populate_public_holidays(self):
         # Israk dan Mikraj
         # Status: In-Use.
 
-        # Isra' and Mi'raj
+        # Isra' and Mi'raj.
         for dt in self._add_isra_and_miraj_day(tr("Israk dan Mikraj")):
             self._add_observed(dt)
 
         # Hari Pertama Berpuasa
         # Status: In-Use.
 
-        # First Day of Ramadan
+        # First Day of Ramadan.
         for dt in self._add_ramadan_beginning_day(tr("Hari Pertama Berpuasa")):
             self._add_observed(dt)
 
         # Hari Nuzul Al-Quran
         # Status: In-Use.
 
-        # Anniversary of the revelation of the Quran
+        # Anniversary of the revelation of the Quran.
         for dt in self._add_nuzul_al_quran_day(tr("Hari Nuzul Al-Quran")):
             self._add_observed(dt)
 
         # Hari Raya Aidil Fitri
         # Status: In-Use.
         # This is celebrate for three days in Brunei.
-        # Observed as 'Hari Raya Puasa' and only for 2 days at certain point.
         # 1: If Wed-Thu-Fri -> Sat (3rd Day +1)
         # 2: If Thu-Fri-Sat -> Mon (2nd Day +3)
-        # 3: If Fri-Sat-Sun -> Mon (1st Day +3)
+        # 3: If Fri-Sat-Sun -> Mon, Tue (1st Day +3, 3rd day +2)
         # 4: If Sat-Sun-Mon -> Tue (2nd Day +2)
         # 5: If Sun-Mon-Tue -> Wed (1st Day +3)
-
-        # Eid al-Fitr
+        # Observed as 'Hari Raya Puasa' and only for 2 days prior to 2012.
+        # 1: If Thu-Fri -> Sat (2nd Day +1)
+        # 2: If Fri-Sat -> Mon (1nd Day +2)
+        # 3: If Sat-Sun -> Mon (2nd Day +1)
+        # 4: If Sun-Mon -> Tue (1st Day +2)
+        # This was only observed for a single day in 2000.
+
+        # Eid al-Fitr.
         name = tr("Hari Raya Aidil Fitri")
-        for dt in self._add_eid_al_fitr_day(name):
-            self._add_observed(dt, rule=FRI_TO_NEXT_MON + SUN_TO_NEXT_WED)
-        for dt in self._add_eid_al_fitr_day_two(name):
-            self._add_observed(dt, rule=FRI_TO_NEXT_MON + SUN_TO_NEXT_TUE)
-        for dt in self._add_eid_al_fitr_day_three(name):
-            self._add_observed(dt, rule=FRI_TO_NEXT_SAT)
+        if self._year >= 2012:
```

Comment:
I think this can be optimized with `...TO_NEXT_WORKDAY`, but I don't see how exactly at the moment.

Reply:
I tried them earlier, and they don't seem to work properly here.